source/dubregistry/repositories/bitbucket.d

```diff
@@ -120,13 +124,21 @@ class BitbucketRepository : Repository {
 		if( ver.startsWith("~") ) ver = ver[1 .. $];
 		else ver = ver;
 		auto venc = () @trusted { return encodeComponent(ver); } ();
+		if (m_authUser.length) return "https://"~m_authUser~":"~m_authPassword~"@bitbucket.org/"~m_owner~"/"~m_project~"/get/"~venc~".zip";
```

Comment:
I think adding another endpoint here (`getContentURLPrefix` for example) would be better in case something changes or another function uses it.

Reply:
Yeah - but YAGNI says otherwise. This is just on function and the URL is contained.

Reply:
Moved the URL to a common place and re-used the common sufix when auth is on. Hope this is OK.